nucypher/blockchain/eth/domains.py

```diff
@@ -0,0 +1,70 @@
+from enum import Enum
+from typing import NamedTuple
+
+
+class ChainInfo(NamedTuple):
+    id: int
+    name: str
+
+
+class EthChain(ChainInfo, Enum):
+    MAINNET = (1, "mainnet")
+    GOERLI = (5, "goerli")
+    SEPOLIA = (11155111, "sepolia")
+    TESTERCHAIN = (131277322940537, "eth-tester")
+
+
+class PolygonChain(ChainInfo, Enum):
+    POLYGON = (137, "polygon")
+    MUMBAI = (80001, "mumbai")
+    TESTERCHAIN = (131277322940537, "eth-tester")
+
+
+class DomainInfo(NamedTuple):
+    name: str
+    eth_chain: EthChain
+    polygon_chain: PolygonChain
+
+    def __str__(self) -> str:
+        return self.name
+
+    def __bytes__(self) -> bytes:
+        return self.name.encode()
+
+    @property
+    def is_testnet(self) -> bool:
+        return self.eth_chain != EthChain.MAINNET
+
+
+class TACoDomain:
+    class Unrecognized(RuntimeError):
+        """Raised when a provided domain name is not recognized."""
+
+    MAINNET = DomainInfo("mainnet", EthChain.MAINNET, PolygonChain.POLYGON)
+    # Testnets
+    LYNX = DomainInfo("lynx", EthChain.GOERLI, PolygonChain.MUMBAI)
+    TAPIR = DomainInfo("tapir", EthChain.SEPOLIA, PolygonChain.MUMBAI)
+    # TODO remove these (oryx, ibex) when appropriate
+    ORYX = DomainInfo("oryx", EthChain.GOERLI, PolygonChain.POLYGON)
+    IBEX = DomainInfo(
+        "ibex", EthChain.GOERLI, None
+    )  # this is required for configuration file migrations (backwards compatibility)
```

Comment:
I think these can be removed - we're not going to support either of these networks post-merge.

```suggestion
```